src/dbnode/storage/stats/query_stats.go

```diff
@@ -105,8 +121,51 @@ func (q *queryStats) Update(newDocs int) error {
 	recentDocs := q.recentDocs.Add(newDocsI64)
 
 	values := QueryStatsValues{
-		RecentDocs: recentDocs,
-		NewDocs:    newDocsI64,
+		DocsMatched: QueryStatsValue{
+			Recent: recentDocs,
+			New:    newDocsI64,
+			Reset:  false,
+		},
+		BytesRead: QueryStatsValue{
+			// Pass current bytes-read along with docs since we want
+			// to check if above that limit as well.
+			Recent: q.recentBytesRead.Load(),
+			New:    0,
+			Reset:  false,
+		},
+	}
+
+	// Invoke the custom tracker based on the new stats values.
+	return q.tracker.TrackStats(values)
+}
+
+// UpdateBytesRead adds new bytes stats to be tracked.
+func (q *queryStats) UpdateBytesRead(newBytesRead int) error {
+	if q == nil {
+		return nil
+	}
+	if newBytesRead < 0 {
+		return nil
+	}
+
+	newBytesReadI64 := int64(newBytesRead)
+
+	// Add the new stats to the global state.
+	recentBytes := q.recentBytesRead.Add(newBytesReadI64)
+
+	values := QueryStatsValues{
+		BytesRead: QueryStatsValue{
+			Recent: recentBytes,
+			New:    newBytesReadI64,
+			Reset:  false,
+		},
+		DocsMatched: QueryStatsValue{
+			// Pass current docs along with bytes-read since we want
```

Comment:
update the comment? seems to be copied...

Reply:
It is slightly different actually from the other comment. I'm trying to callout w/ those two comments that in `UpdateDocs` we still check the bytes-read limit (and vice versa).